Neos.Flow/Classes/Session/Aspect/LoggingAspect.php

```diff
@@ -81,76 +58,58 @@ public function logResume(JoinPointInterface $joinPoint)
             } else {
                 $inactivityMessage = sprintf('more than %s hours', (int)($inactivityInSeconds / 3600));
             }
-            $this->logger->debug(sprintf('%s: Resumed session with id %s which was inactive for %s. (%ss)', $this->getClassName($joinPoint), $session->getId(), $inactivityMessage, $inactivityInSeconds));
+            $this->logger?->debug(sprintf('%s: Resumed session with id %s which was inactive for %s. (%ss)', $this->getSessionImplementationClassName($joinPoint), $session->getId(), $inactivityMessage, $inactivityInSeconds));
         }
     }
 
     /**
-     * Logs calls of destroy()
-     *
-     * @Flow\Before("within(Neos\Flow\Session\SessionInterface) && method(.*->destroy())")
-     * @param JoinPointInterface $joinPoint The current joinpoint
-     * @return void
+     * @throws SessionNotStartedException
      */
-    public function logDestroy(JoinPointInterface $joinPoint)
+    #[Flow\Before("within(Neos\Flow\Session\SessionInterface) && method(.*->destroy())")]
+    public function logDestroy(JoinPointInterface $joinPoint): void
     {
-        /** @var SessionInterface $session */
         $session = $joinPoint->getProxy();
-        if ($session->isStarted()) {
+        if ($session instanceof SessionInterface && $session->isStarted()) {
             $reason = $joinPoint->isMethodArgument('reason') ? $joinPoint->getMethodArgument('reason') : 'no reason given';
-            $this->logger->debug(sprintf('%s: Destroyed session with id %s: %s', $this->getClassName($joinPoint), $session->getId(), $reason));
+            $this->logger?->debug(sprintf('%s: Destroyed session with id %s: %s', $this->getSessionImplementationClassName($joinPoint), $session->getId(), $reason));
         }
     }
 
     /**
-     * Logs calls of renewId()
-     *
-     * @Flow\Around("within(Neos\Flow\Session\SessionInterface) && method(.*->renewId())")
-     * @param JoinPointInterface $joinPoint The current joinpoint
-     * @return mixed The result of the target method if it has not been intercepted
+     * @throws SessionNotStartedException
      */
-    public function logRenewId(JoinPointInterface $joinPoint)
+    #[Flow\Around("within(Neos\Flow\Session\SessionInterface) && method(.*->renewId())")]
+    public function logRenewId(JoinPointInterface $joinPoint): mixed
     {
-        /** @var SessionInterface $session */
         $session = $joinPoint->getProxy();
-        $oldId = $session->getId();
         $newId = $joinPoint->getAdviceChain()->proceed($joinPoint);
-        if ($session->isStarted()) {
-            $this->logger->info(sprintf('%s: Changed session id from %s to %s', $this->getClassName($joinPoint), $oldId, $newId));
+        if ($session instanceof SessionInterface) {
+            $oldId = $session->getId();
```

Comment:
Doesn't look like we need $oldId outside the next "if", so this could again be a combination instanceof & isStarted "if" here?

Reply:
Will fix.